include/socgraph.php

```diff
@@ -190,181 +187,166 @@ function poco_load_worker($cid, $uid, $zcid, $url) {
 	);
 
 }
+/**
+ * @brief Sanitize the given gcontact data
+ *
+ * @param array $gcontact array with gcontact data
+ *
+ * Generation:
+ *  0: No definition
+ *  1: Profiles on this server
+ *  2: Contacts of profiles on this server
+ *  3: Contacts of contacts of profiles on this server
+ *  4: ...
+ *
+ */
+function sanitized_gcontact(&$gcontact) {
```

Comment:
I must confess that this exception stuff seems to be bloating up the code, which I don't like. But I also must confess that I understand your concerns to the calling by reference.

Do we always need to add this "catch" thing to the exception or can someone™ create some exception class that would automatically add logging in case of a catch and that would avoid us from the need to add this to every "try" construct?